modules/redpanda/redpanda.go

```diff
@@ -252,6 +260,29 @@ func renderBootstrapConfig(settings options) ([]byte, error) {
 	return bootstrapConfig.Bytes(), nil
 }
 
+// registerListeners validates that the provided listeners are valid and set network aliases for the provided addresses.
+// The container must be attached to at least one network.
+func registerListeners(ctx context.Context, settings options, req testcontainers.GenericContainerRequest) error {
+	if len(settings.Listeners) == 0 {
+		return nil
+	}
+
+	if len(req.Networks) == 0 {
+		return fmt.Errorf("container must be attached to at least one network")
+	}
+
+	for _, listener := range settings.Listeners {
+		if listener.Port < 0 || listener.Port > math.MaxUint16 {
+			return fmt.Errorf("invalid port on listener %s:%d (must be between 0 and 65535)", listener.Address, listener.Port)
+		}
+
+		for _, network := range req.Networks {
+			req.NetworkAliases[network] = append(req.NetworkAliases[network], listener.Address)
+		}
```

Comment:
one issue with this approach that Java implementation still have is that when adding Toxiproxy upfront Kafka container and using `0.0.0.0:<random-toxiproxy-port>` as a listener the registration will fail because will try to add 0.0.0.0 as an alias. I forgot to mention that when suggesting the approach initially.

Reply:
@sago2k8 @eddumelendez would you like to collaborate in a follow-up PR for this?

Reply:
Hey, thanks for your comments, yes I could work on it, do you happen to have a suggestion of a expected behaviour for that specific case